test/integration/dualstack/dualstack_endpoints_test.go

```diff
@@ -0,0 +1,294 @@
+		_, err := client.CoreV1().Services(metav1.NamespaceDefault).Get(context.TODO(), "kubernetes", metav1.GetOptions{})
+		if err != nil && !apierrors.IsNotFound(err) {
+			return false, err
+		}
+		return !apierrors.IsNotFound(err), nil
+	}); err != nil {
+		t.Fatalf("creating kubernetes service timed out")
+	}
+
+	resyncPeriod := 0 * time.Hour
+	informers := informers.NewSharedInformerFactory(client, resyncPeriod)
+
+	// Create fake node
+	testNode := &v1.Node{
+		ObjectMeta: metav1.ObjectMeta{
+			Name: "fakenode",
+		},
+		Spec: v1.NodeSpec{Unschedulable: false},
+		Status: v1.NodeStatus{
+			Conditions: []v1.NodeCondition{
+				{
+					Type:              v1.NodeReady,
+					Status:            v1.ConditionTrue,
+					Reason:            fmt.Sprintf("schedulable condition"),
+					LastHeartbeatTime: metav1.Time{Time: time.Now()},
+				},
+			},
+		},
+	}
+	if _, err := client.CoreV1().Nodes().Create(context.TODO(), testNode, metav1.CreateOptions{}); err != nil {
+		t.Fatalf("Failed to create Node %q: %v", testNode.Name, err)
+	}
+
+	epController := endpoint.NewEndpointController(
+		informers.Core().V1().Pods(),
+		informers.Core().V1().Services(),
+		informers.Core().V1().Endpoints(),
+		client,
+		1*time.Second)
+
+	epsController := endpointslice.NewController(
+		informers.Core().V1().Pods(),
+		informers.Core().V1().Services(),
+		informers.Core().V1().Nodes(),
+		informers.Discovery().V1beta1().EndpointSlices(),
+		int32(100),
+		client,
+		1*time.Second)
+
+	// Start informer and controllers
+	stopCh := make(chan struct{})
+	defer close(stopCh)
+	informers.Start(stopCh)
+	// use only one worker to serialize the updates
+	go epController.Run(1, stopCh)
+	go epsController.Run(1, stopCh)
+
+	var testcases = []struct {
+		name           string
+		serviceType    v1.ServiceType
+		ipFamilies     []v1.IPFamily
+		ipFamilyPolicy v1.IPFamilyPolicyType
+	}{
+		{
+			name:           "Service IPv4 Only",
+			serviceType:    v1.ServiceTypeClusterIP,
+			ipFamilies:     []v1.IPFamily{v1.IPv4Protocol},
+			ipFamilyPolicy: v1.IPFamilyPolicySingleStack,
+		},
+		{
+			name:           "Service IPv6 Only",
+			serviceType:    v1.ServiceTypeClusterIP,
+			ipFamilies:     []v1.IPFamily{v1.IPv6Protocol},
+			ipFamilyPolicy: v1.IPFamilyPolicySingleStack,
+		},
+		{
+			name:           "Service IPv6 IPv4 Dual Stack",
+			serviceType:    v1.ServiceTypeClusterIP,
+			ipFamilies:     []v1.IPFamily{v1.IPv6Protocol, v1.IPv4Protocol},
+			ipFamilyPolicy: v1.IPFamilyPolicyRequireDualStack,
+		},
+		{
+			name:           "Service IPv4 IPv6 Dual Stack",
+			serviceType:    v1.ServiceTypeClusterIP,
+			ipFamilies:     []v1.IPFamily{v1.IPv4Protocol, v1.IPv6Protocol},
+			ipFamilyPolicy: v1.IPFamilyPolicyRequireDualStack,
+		},
+	}
+
+	for i, tc := range testcases {
+		t.Run(tc.name, func(t *testing.T) {
+			ns := framework.CreateTestingNamespace(fmt.Sprintf("test-endpointslice-dualstack-%d", i), s, t)
+			defer framework.DeleteTestingNamespace(ns, s, t)
```

Comment:
note, create returns a NS object but does not actually create it using a client. also delete is a no-op:
https://github.com/kubernetes/kubernetes/blob/master/test/integration/framework/util.go#L48-L64

Reply:
@sttts  kindly helped me with this, it seems that the integration framework does not check for the existance of namespaces, however, the tests uses the right URLs

> I1027 11:11:40.380932   70653 request.go:581] Throttling request took 196.832197ms, request: PUT:http://127.0.0.1:35039/api/v1/namespaces/test-endpointslice-dualstack-3/pods/test-pod/status
I1027 11:11:40.383711   70653 httplog.go:89] "HTTP" verb="PUT" URI="/api/v1/namespaces/test-endpointslice-dualstack-3/pods/test-pod/status" latency="2.35047ms" userAgent="dualstack.test/v0.0.0 (linux/amd64) kubernetes/$Format" srcIP="127.0.0.1:45852" resp=200

this pattern is used in other test, so I would left as "Work as expected" :smile: